### DISCUSSION TITLE
Auto-Increament Numeric Id

### DISCUSSION THREAD

**Discussion by sushruthkumar123** [IEvangelist/azure-cosmos-dotnet-repository#237] [Ideas]
Auto increament numeric id sequencially when creating a new document of the same "_type" .

**Comment by sushruthkumar123** [comment]
So, we are trying to migrate from sql server to cosmos and we want to previous the old pk and start creating new sequential ids when creating new documents in cosmos db.

**Comment by IEvangelist** [comment]
@mumby0168 - this idea for auto-incrementing id's could be done with stored-procedures. For a given type, when it is created, within the stored-procedure it could get the max id, and add one for the current write.

**Reply by mumby0168** [reply]
yeah, that is about the only way I think

**Reply by sushruthkumar123** [reply]
Not sure if it will work for documents of same type but with different patition key. Since "Stored procedures are always scoped to a partition key"
